packages/common/src/index-api.ts

```diff
@@ -24,13 +24,75 @@ export type BackwardPagination = {
  */
 export type Pagination = ForwardPagination | BackwardPagination
 
+/**
+ * Boolean field value filter
+ */
+export type BooleanValueFilter = { isNull: boolean } | { equalTo: boolean }
+
+/**
+ * Enum field value filter
+ */
+export type EnumValueFilter =
+  | { isNull: boolean }
+  | { equalTo: string }
+  | { notEqualTo: string }
+  | { in: Array<string> }
+  | { notIn: Array<string> }
+
+/**
+ * String or number field value filter
+ */
+export type ScalarValueFilter<T extends string | number = string | number> =
+  | { isNull: boolean }
+  | { equalTo: T }
+  | { notEqualTo: T }
+  | { in: Array<T> }
+  | { notIn: Array<T> }
+  | { lessThan: T }
+  | { lessThanOrEqualTo: T }
```

Comment:
Similar to not, we have an or filter, so this seems redundant.

Reply:
When it comes to usability, I think it's much easier to provide and document the ability to do ex:
```js
{ doc: { date: { greaterThanOrEqualTo: '2023-06-01' }, status: { notEqualTo: 'ARCHIVED' } } }
```
than the alternative without support for these:
```js
{ and: [
  { or: [
    { doc: { date: equalTo: '2023-06-01' } },
    { doc: { date: greaterThan: '2023-06-01' },
  }]},
  { not: {
    doc: { status: { equalTo: 'ARCHIVED' } }
  }}]
}
```
This would be much more in line with what similar systems are providing.

Reply:
2 cents. I agree _this_ reads nicer. As with other similar systems though, it is a pain (for me!) to express anything more complicated than a hello world example.